Cython/Compiler/Code.py

```diff
@@ -1400,9 +1404,18 @@ def generate_object_constant_decls(self):
                   for c in self.py_constants]
         consts.sort()
         decls_writer = self.parts['decls']
+        decls_writer.putln("#if !CYTHON_COMPILING_IN_LIMITED_API")
         for _, cname, c in consts:
+            self.parts['module_state'].putln("%s;" % c.type.declaration_code(cname))
+            self.parts['module_state_defines'].putln(
+                "#define %s %s->%s" % (cname, Naming.modulestateglobal_cname, cname))
+            self.parts['module_state_clear'].putln(
+                "Py_CLEAR(%s(m)->%s);" % (Naming.modulestate_cname, cname))
```

Comment:
BTW, `Py_CLEAR()` is a macro that evaluates the argument twice. Is that an issue here?

Reply:
> Or rather, we could assign the module state pointer to a local variable at the beginning of the module cleanup function

That sounds about right, let me add that!

Reply:
Done